test/tools/ossfuzz/abiV2FuzzerCommon.h

```diff
@@ -0,0 +1,88 @@
+#pragma once
+
+#include <libsolidity/interface/CompilerStack.h>
+#include <libyul/AssemblyStack.h>
+#include <liblangutil/Exceptions.h>
+#include <liblangutil/SourceReferenceFormatter.h>
+#include <libdevcore/Keccak256.h>
+
+namespace dev
+{
+namespace test
+{
+namespace abiv2fuzzer
+{
+class SolidityExecutionFramework
+{
+public:
+	SolidityExecutionFramework();
+	SolidityExecutionFramework(langutil::EVMVersion _evmVersion);
+
+	Json::Value getMethodIdentifiers()
+	{
+		return m_compiler.methodIdentifiers(m_compiler.lastContractName());
+	}
+
+	dev::bytes compileContract(
+			std::string const& _sourceCode,
+			std::string const& _contractName = "",
+			std::map<std::string, dev::FixedHash<20>> const& _libraryAddresses = std::map<std::string, dev::FixedHash<20>>()
+	)
+	{
+		// Silence compiler version warning
+		std::string sourceCode = "pragma solidity >=0.0;\n";
+		if (_sourceCode.find("pragma experimental ABIEncoderV2;") == std::string::npos)
+			sourceCode += "pragma experimental ABIEncoderV2;\n";
+		sourceCode += _sourceCode;
+		m_compiler.reset();
+		m_compiler.setSources({{"", sourceCode}});
+		m_compiler.setLibraries(_libraryAddresses);
+		m_compiler.setEVMVersion(m_evmVersion);
+		m_compiler.setOptimiserSettings(m_optimiserSettings);
```

Comment:
Should probably activate everything, i.e. `OptimiserSettings::full()`

Reply:
done